scripts/dataset_processing/tts/audio_processing/preprocess_audio.py

```diff
@@ -99,53 +136,51 @@ def _process_entry(
     if volume_level:
         audio = normalize_volume(audio, volume_level=volume_level)
 
+    output_path.parent.mkdir(exist_ok=True, parents=True)
     sf.write(file=output_path, data=audio, samplerate=sample_rate)
 
-    original_duration = librosa.get_duration(filename=str(audio_filepath))
+    original_duration = librosa.get_duration(filename=str(audio_path))
     output_duration = librosa.get_duration(filename=str(output_path))
 
-    entry["audio_filepath"] = output_path
+    entry["audio_filepath"] = str(output_audio_filepath)
```

Comment:
nit: this function actually applied many forced conversions from `pathlib.Path` to `str`. I feel it is redundant. Maybe we can only specify a `str` type for `input_audio_dir` and `output_audio_dir`?

Reply:
looks good to me. Thanks.